base_layer/core/src/mining/miner.rs

```diff
@@ -293,3 +323,12 @@ impl<B: BlockchainBackend> Miner<B> {
         Ok(())
     }
 }
+
+async fn check_kill_flag(kill_flag: Arc<AtomicBool>, stop_flag: Arc<AtomicBool>) {
+    loop {
+        tokio::time::delay_for(Duration::from_millis(100)).await;
```

Comment:
Spin waiting is an unnecessary cost - a request stream piping instructions to a mining actor would remove the need to do this

Reply:
Yip, that function should be thrown out as soon as possible. This just interprets the current base node kill signal